snapshot.go

```diff
@@ -5,17 +5,10 @@ import "C"
 
 // Snapshot provides a consistent view of read operations in a DB.
 type Snapshot struct {
-	c   *C.rocksdb_snapshot_t
-	cDb *C.rocksdb_t
+	c *C.rocksdb_snapshot_t
 }
 
 // NewNativeSnapshot creates a Snapshot object.
-func NewNativeSnapshot(c *C.rocksdb_snapshot_t, cDb *C.rocksdb_t) *Snapshot {
-	return &Snapshot{c, cDb}
-}
-
-// Release removes the snapshot from the database's list of snapshots.
-func (s *Snapshot) Release() {
-	C.rocksdb_release_snapshot(s.cDb, s.c)
-	s.c, s.cDb = nil, nil
+func NewNativeSnapshot(c *C.rocksdb_snapshot_t) *Snapshot {
```

Comment:
If you absolutely prefer a Destroyer, I can implement it. Just wanted to give my two cents here.